javascript-style-guide.md

```diff
@@ -143,6 +179,131 @@ algorithms that might grow the call stack boundlessly.
 Aggregate operations provide forms for iterating over collections that are more
 precise and oftentimes more useful than their procedural alternatives.
 
+## Object-orientation
+
+You can leverage object-oriented design patterns such as encapsulation,
+composition, and inheritance in JavaScript. Create functions whose instance
+variables represent private and protected members. From these functions, return
+objects with methods to access and possibly manipulate the instance
+variables. Create functions with extended interfaces that mutate instances of
+their parents to effect inheritance.
+
+Example:
+
+```js
+var makeMammal = function (data) {
+    var self = {};
+    self.getName = function () {
+        return data.name;
+    };
+    self.says = function () {
+        return data.saying === undefined ? '' : data.saying;
+    };
+    return self;
+};
+
+var myMammal = makeMammal({name: 'Herb'});
+
+var makeCat = function (data) {
+    var self;
+    data.saying = data.saying === undefined ? 'meow' : data.saying;
+    self = makeMammal(data);
+    self.purr = function (n) {
+        var message = '',
+            index;
+        for (index = 0; index < n; index += 1) {
+            if (message.length > 0) {
+                message += '-';
+            }
+            message += 'r';
+        }
+        return message;
+    };
+    self.getName = function () {
+        return self.says() + ' ' + data.name +
+            ' ' + self.says();
+    };
+    return self;
+};
+
+var myCat = makeCat({name: 'Henrietta'});
+```
+
+## Existence
+
+In JavaScript, the values `false`, `undefined` and `null` are falsy, but so are
+`''` and `0`. This is unfortunate, because sometimes empty strings and zero are
+"valid" values from a business perspective. Therefore, when checking for the
+"existence" of a value, it is unreliable to rely on truthiness or
+falsiness. Instead, you should explicitly check for negative values.
```

Comment:
Better to say *bottom values* instead of *negative values*. [That's what Crockford would say.](https://youtu.be/hQVTIJBZook?t=887)